### PR TITLE
#10 db data structure - dev

### PR DESCRIPTION
**Pull request by JanefrancessC**
fixing issue #10 

Created a test Database Schema and added a document. The .get was also activated to display the contents of the database.

### CODE REVIEW

**Review by Sharifa1 — APPROVED** [review]
Well done Chioma. You have done a great job. For me all you've done so far looks okay.

**Review by Sharifa1 — APPROVED** [review]
Well done Chioma. You have done a great job. For me all you've done so far looks okay.

File: server/controllers/video.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        // const newVideo = new Video ({
+        //     title: "Lost love" ,
+        //     desc: "Learn English through stories - Level 1",
+        //     videoUrl:"https://youtu.be/qtHy_DmOmLE",
+        //     views: 5,
+        // })
+        await newVideo.save();
+        res.status(200).json("Video successfully added")
+    } catch (error) {
+        next(error);
```
Comment (PakizeBozkurt) [inline server/controllers/video.js:17]: How about thinking meaningful error messages when an error occurs:
catch (error) {
  res.status(500).json({ success: false, error: 'Failed to create video' });
}

Comment (PakizeBozkurt) [inline server/controllers/video.js:17]: How about thinking meaningful error messages when an error occurs:
catch (error) {
res.status(500).json({ success: false, error: 'Failed to create video' });
}

File: server/controllers/user.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export const test = (req, res) => {
+    res.json('Test works');
```
Comment (PakizeBozkurt) [inline server/controllers/user.js:2]: Great to have nice test!
Thank you Chioma 😺

File: server/controllers/video.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import mongoose from 'mongoose';
+import Video  from '../models/Video.js'
+
+export const testVideo = async (req,res, next) => {
```
Comment (PakizeBozkurt) [inline server/controllers/video.js]: About function names: can we say 'createVideo'?

File: server/index.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import express from "express";
+import mongoose from "mongoose";
+import * as dotenv from "dotenv";
```
Comment (PakizeBozkurt) [inline server/index.js:3]: Do you think we need to use '*' or
import dotenv from 'dotenv'; ?

File: server/models/User.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    email: {
+        type: String,
+        unique: true,
+        required: [true, "Email is required"],
+    },
+    password: {
+        type: String,
+        required: true,
+    }
+},
```
Comment (PakizeBozkurt) [inline server/models/User.js:18]: Thank you for defining a mongoose schema and model for a User entity. Well done!

File: server/models/Video.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        default: [],
+    },
+    dislikes: {
+        type: [String],
+        default: [],
+    }
+}, 
+{ timestamps: true });
+
+
```
Comment (PakizeBozkurt) [inline server/models/Video.js:34]: 👏🏻 👍🏻 Thank you!

File: server/routes/beginners-level.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import express from 'express';
+import { displayVideo, testVideo } from '../controllers/video.js';
+// import {  } from '../controllers/video.js';
+
+const videoRouter = express.Router();
+videoRouter.get('/beginners-level', displayVideo); 
+videoRouter.post('/beginners-level', testVideo); 
+
```
Comment (PakizeBozkurt) [inline server/routes/beginners-level.js:8]: The code looks good and follows the standard practices for defining routes with Express.js and assigning route handlers. Great job!

**Review by PakizeBozkurt — APPROVED** [review]
Fantastic job Chioma!
I left some messages to rethink.
Thank you!

File: server/controllers/video.js (hunk 3 above)
Comment (JanefrancessC) [reply server/controllers/video.js]: Okay. I'm going to implement that

> About function names: can we say 'createVideo'?

File: server/controllers/video.js (hunk 1 above)
Comment (JanefrancessC) [reply server/controllers/video.js:17]: I've told you about the error handling

File: server/index.js (hunk 4 above)
Comment (JanefrancessC) [reply server/index.js:3]: dotenv works fine, but the * as dotenv is what is mostly used in the industry. That's why i used that

File: server/models/User.js (hunk 5 above)
Comment (JanefrancessC) [reply server/models/User.js:18]: It's still subject to change. as I'm not yet sure about the fields we will have for a user

File: server/routes/beginners-level.js (hunk 7 above)
Comment (JanefrancessC) [reply server/routes/beginners-level.js:8]: Thank you Pakize.

**Pull request merged**